blockchain/blockchain.go

```diff
@@ -2790,6 +2810,28 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 	return receipt, internalTrace, err
 }
 
+func (bc *BlockChain) RegisterExecutionModule(modules ...kaiax.ExecutionModule) {
+	bc.executionModules = append(bc.executionModules, modules...)
+}
+
+func (bc *BlockChain) RegisterRewindableModule(modules ...kaiax.RewindableModule) {
+	bc.rewindableModules = append(bc.rewindableModules, modules...)
+}
+
+// Stop modules before performing a SetHead while node is running, i.e. via debug_setHead.
+func (bc *BlockChain) StopRewindableModules() {
```

Comment:
Could not imagine how it is useful. Can you please share how it forms?

Reply:
There is exactly one place it's needed, the debug_setHead API.

In CNAPIBackend.SetHead and doSetHead (https://github.com/kaiachain/kaia/blob/v1.0.2/node/cn/api_backend.go#L84-106), several components/potential modules are restarted or their cache flushed.

Maybe I should create a separate callback, (not Start(), Stop()) for this purpose?
like PreRewind() and PostRewind()?

Reply:
Yes maybe, that would be more intuitive.

The code below is initially difficult to interpret. Renaming `stop` and `start` to something like `pre` and `post` could make it easier to understand their purpose.
```
...
b.cn.blockchain.(kaiax.RewindableModuleHost).StopRewindableModules()
defer b.cn.blockchain.(kaiax.RewindableModuleHost).StartRewindableModules()
...
```